src/part_table.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 Red Hat, Inc.
+
+// Licensed under the MIT license <LICENSE or
+// http://opensource.org/licenses/MIT> This file may not be copied, modified,
+// or distributed except according to those terms.
+
+use crate::{error::c_result, partition::Partition, BlkIdResult};
+use blkid_sys::*;
+use std::{ffi::CStr, str::FromStr};
+use strum_macros::{Display, EnumString};
+
+/// Information about a partition table
+#[derive(Debug)]
+pub struct PartTable(pub(crate) blkid_parttable);
+
+impl PartTable {
+    /// Returns partition table ID (for example GPT disk UUID).
+    ///
+    /// The ID is GPT disk UUID or DOS disk ID (in hex format).
+    pub fn get_id(&self) -> Option<String> {
+        let ptr = unsafe { blkid_parttable_get_id(self.0) };
+        if ptr.is_null() {
+            None
+        } else {
+            Some(unsafe { CStr::from_ptr(ptr).to_string_lossy().to_string() })
+        }
+    }
+
+    /// Returns position (in bytes) of the partition table.
+    ///
+    /// # NOTE
+    ///
+    /// The position is relative to begin of the device as defined by `Prober::set_device` for
+    /// primary partition table, and relative to parental partition for nested partition tables.
+    pub fn get_offset(&self) -> BlkIdResult<i64> {
+        unsafe { c_result(blkid_parttable_get_offset(self.0)) }
+    }
+
+    /// Returns parent for nested partition tables
+    pub fn get_parent(&self) -> Option<Partition> {
+        let part = unsafe { blkid_parttable_get_parent(self.0) };
+        if part.is_null() {
+            None
+        } else {
+            Some(Partition(part))
+        }
+    }
+
+    /// Returns partition table type (type name, e.g. "dos", "gpt", ...)
+    pub fn get_type(&self) -> Option<PartitionTableType> {
+        let ptr = unsafe { blkid_parttable_get_type(self.0) };
+        if ptr.is_null() {
+            None
+        } else {
+            let part_table_type = unsafe { CStr::from_ptr(ptr).to_string_lossy() };
+            let part_table_type = PartitionTableType::from_str(part_table_type.as_ref())
+                .expect("BUG: strum is broken, it must use default");
```

Comment:
It might be better to return a `Result` here instead of a panic. Since this is a library the caller of the library cannot recover from this panic if it's encountered. If this returns a `Result` instead though they can do something about it.

Reply:
I'm sure this never panic because if you look at enum you can see that it has stum default attribute and if it cannot to convert string in any available variants it will use default field. So, we don't need to pollute code with wrapping option into Result. Or if you care about it we can just return None instead of use default (or explicitly build Unknown option).

Reply:
Fixed.